src/consensus/blockstore/slot_block_data.rs

```diff
@@ -217,19 +217,32 @@ impl BlockData {
     /// Returns `true` if a slice was reconstructed, `false` otherwise.
     fn try_reconstruct_slice(&mut self, slice: usize) -> bool {
         let slice_shreds = self.shreds.get(&slice).unwrap();
-        if slice_shreds.len() < shredder::DATA_SHREDS || self.slices.contains_key(&slice) {
+        if self.slices.contains_key(&slice) {
+            return false;
+        }
+        let reconstructed_slice = match RegularShredder::deshred(slice_shreds) {
+            Ok(s) => s,
+            Err(DeshredError::NotEnoughShreds) => return false,
+            rest => {
+                warn!("deshreding failed with {rest:?}");
+                return false;
```

Comment:
Question: in this case and the case where the parent is not present below, shouldn't we stop receiving additional shreds for this slice as we should now have sufficient evidence that we will never properly reconstruct it.

Reply:
We could probably handle it in subsequent PRs.

Reply:
Exactly, what we can even do is treat it the same as equivocation right now, so tigger `try_skip_window()` to vote to skip the leader